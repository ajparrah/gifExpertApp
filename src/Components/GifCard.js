import React from 'react'

export const GifCard = ({image}) => {
    
    return (
        <div className="col-4 mt-2 animate__animated animate__fadeIn">
           <div className="card text-center">
                <img src={image.url} className="card-img-top" alt={image.title} style={{width: '100%', height: '200px'}}/>
                <div className="card-body" style={{height: '100px'}}>
                    <h5 className="card-title">{image.title}</h5>
                </div>
            </div> 
        </div>
        
    )
}
