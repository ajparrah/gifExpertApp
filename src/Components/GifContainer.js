import React from 'react';
import { useFetchGifs } from '../Hooks/useFetchGifs';
import { GifCard } from './GifCard';
import { Spinner } from './Spinner/Spinner';

export const GifContainer = ({category}) => {
    const {data: images, loading} =  useFetchGifs(category);
    const renderCards = () => {
        return images.map((img) => (
            <GifCard
                key={img.id}
                image={img}
            />
        ))
    }
    return (
        <div className="row mb-3">
            {
                loading ? <Spinner/> : renderCards()
            }
        </div>
    )
}
