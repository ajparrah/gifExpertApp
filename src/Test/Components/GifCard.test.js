import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifCard } from '../../Components/GifCard';

describe('Pruebas en <GifCard />', () => {
  // #region Component and props required
  const image = {
    title: 'Imagen de prueba',
    url: 'https://localhost:8000'
  };
  const wrapper = shallow(<GifCard image={image}/>);
  // #endregion

  test('Debe mostrar correctamente el componente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe poseer un titulo la imagen', () => {
    const titleElement = wrapper.find('.card-title');
    expect(titleElement.text().trim()).toBe(image.title);
  });

  test('Debe poseer un titulo alternativo y URL valida la imagen', () => {
    const imgElement = wrapper.find('img');
    //.props, retorna un objeto con las propiedades del elemento, como por ejemplo el src, alt...
    //.prop, recibe el identificador de la propiedad, como por ejemplo. .prop('src'). Busca el atributo del elemento para validar
    expect(imgElement.prop('src').trim()).toBe(image.url);
    expect(imgElement.prop('alt').trim()).toBe(image.title);
  });

  describe('Debe poseer la clase fadeIn para que sea animado al renderizar', () => {
    const divElement = wrapper.find('#gifCard');
    test('Debe poseer la clase animated por ser requerido por la libreria', () => {
      expect(divElement.hasClass('animate__animated')).toBe(true); //Requerido para que sea posible la animacion
    });

    test('Debe poseer la clase fadeIn', () => {
      expect(divElement.hasClass('animate__fadeIn')).toBe(true);  
    });
  });
  
  
});
