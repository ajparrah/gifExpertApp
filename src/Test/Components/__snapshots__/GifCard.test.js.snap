// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifCard /> Debe mostrar correctamente el componente 1`] = `
<div
  className="col-12 col-md-6 col-lg-4 mt-2 animate__animated animate__fadeIn"
  id="gifCard"
>
  <div
    className="card text-center"
  >
    <img
      alt="Imagen de prueba"
      className="card-img-top"
      src="https://localhost:8000"
      style={
        Object {
          "height": "200px",
          "width": "100%",
        }
      }
    />
    <div
      className="card-body"
      style={
        Object {
          "height": "7em",
        }
      }
    >
      <h5
        className="card-title"
        id="title"
      >
        Imagen de prueba
      </h5>
    </div>
  </div>
</div>
`;
